.github/workflows/apps/output_supported_versions.go

```diff
@@ -0,0 +1,293 @@
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"sort"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+	"path/filepath"
+	"github.com/DataDog/dd-trace-go/v2/instrumentation"
+)
+
+type ModuleVersion struct {
+	Name        string
+	MinVersion  string
+	MaxVersion  string
+	Repository  string
+	isInstrumented bool
+
+}
+
+type Integration struct {
+	Name       string `yaml:"name"`
+	Repository string `yaml:"repository"`
+}
+
+func isSubdirectory(url, pattern string) bool {
+	if strings.HasPrefix(url, pattern) {
+		// match is either exact or followed by a "/"
+		return len(url) == len(pattern) || url[len(pattern)] == '/'
+	}
+	return false
+}
+
+func fetchLatestVersion(module string) (string, error) {
+	// Fetches latest version with `go list`
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "go", "list", "-m", "-versions", module)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if ctx.Err() == context.DeadlineExceeded {
+		return "", fmt.Errorf("command timed out")
+	}
+	if err != nil {
+		return "", fmt.Errorf("command error: %s", stderr.String())
+	}
+
+	versions := strings.Fields(stdout.String())
+	if len(versions) < 1 {
+		return "", fmt.Errorf("no versions found for module: %s", module)
+	}
+
+	return versions[len(versions)-1], nil
+}
+func isModuleInstrumented(moduleName string, instrumentedSet map[string]struct{}) bool {
+	// whether the module has automatic tracing supported (by Orchestrion)
+	// _, isInstrumented := instrumentedSet[moduleName]
+	isInstrumented := false
+	for key := range instrumentedSet {
+		if isSubdirectory(moduleName, key) {
+			isInstrumented = true
+			break
+		}
+	}
+	return isInstrumented
+}
+
+// parses the go.mod file for a package and extracts the version of a given repository.
+func GetMinVersion(packageName, repositoryName string) (ModuleVersion, error) {
+	// Path to contrib/{packageName}
+	contribPath := filepath.Join("contrib", packageName)
+	goModPath := filepath.Join(contribPath, "go.mod")
+
+	// Check if go.mod exists in directory
+	if _, err := os.Stat(goModPath); os.IsNotExist(err) {
+		return ModuleVersion{}, fmt.Errorf("go.mod not found in %s", contribPath)
+	}
+
+	// Open and parse the go.mod file
+	file, err := os.Open(goModPath)
+	if err != nil {
+		return ModuleVersion{}, fmt.Errorf("failed to open go.mod: %w", err)
+	}
+	defer file.Close()
+
+	requireRegex := regexp.MustCompile(`^\s*([^\s]+)\s+v([0-9]+\.[0-9]+\.[0-9]+.*)`)
+
+	scanner := bufio.NewScanner(file)
+	inRequireBlock := false
+
+	var minVersion string
+
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		if strings.HasPrefix(line, "require (") {
+			inRequireBlock = true
+			continue
+		}
+
+		if inRequireBlock && line == ")" {
+			inRequireBlock = false
+			continue
+		}
+
+		// Check for inline require statement (not in a block)
+		if !inRequireBlock && strings.HasPrefix(line, "require ") {
```

Comment:
you can use https://pkg.go.dev/golang.org/x/mod/modfile to parse the go mod file in an easier way:

```go
data, err := os.ReadFile("/tmp/dat")
if err != nil {
	...
}
f, err := modfile.Parse("go.mod", data, nil)
if err != nil {
	...
}

// use f.Require contains all the go.mod dependencies.
```